src/main/java/use_case/execute_buy/ExecuteBuyInteractor.java

```diff
@@ -74,9 +79,22 @@ public void execute(ExecuteBuyInputData data) {
             outputPresenter.prepareStockNotFoundExceptionView();
         } catch (InsufficientBalanceException e) {
             outputPresenter.prepareInsufficientBalanceExceptionView();
+        } catch (InvalidQuantityException e) {
+            outputPresenter.prepareInvalidQuantityExceptionView(e.getMessage());
         }
     }
 
+    /**
+     * This method checks if the user has sufficient balance to buy the stock.
+     *
+     * @param user      the user
+     * @param totalCost the total cost
+     * @return true if the user has sufficient balance, false otherwise
+     */
+    private boolean isBalanceSufficient(User user, double totalCost) {
```

Comment:
Since the condition checking is only one line, there is no need to create a helper function. Just put the condition inside the `If` statement.

Reply:
I think I didn't merge the latest dev into my branch, and have conflicts

Reply:
In this case, please merge the latest dev branch and resolve merge conflicts locally.